src/Microsoft.AspNetCore.Routing/RouteValuesAddressScheme.cs

```diff
@@ -93,21 +112,24 @@ private void BuildOutboundMatches()
         ///     requiredValues: new { controller = "Orders", action = "GetById" },
         ///   A call to GetLink("OrdersApi", new { id = "10" }) cannot generate url as neither the supplied values or
         ///   current ambient values do not satisfy the decision tree that is built based on the required values.
-        protected virtual (IEnumerable<OutboundMatch>, IDictionary<string, List<OutboundMatchResult>>) GetOutboundMatches()
+        protected virtual (List<OutboundMatch>, Dictionary<string, List<OutboundMatchResult>>) GetOutboundMatches()
         {
             var allOutboundMatches = new List<OutboundMatch>();
             var namedOutboundMatchResults = new Dictionary<string, List<OutboundMatchResult>>(
                 StringComparer.OrdinalIgnoreCase);
 
-            var endpoints = _dataSource.Endpoints.OfType<RouteEndpoint>();
-            foreach (var endpoint in endpoints)
+            foreach (var endpoint in _dataSource.Endpoints)
             {
+                var routeEndpoint = endpoint as RouteEndpoint;
```

Comment:
nit: missing braces from if

```suggestion
                if (!(endpoint is RouteEndpoint routeEndpoint))
                {
                    continue;
                }
```

Reply:
Done.